src/services/ProductTransactions.ts

```diff
@@ -40,12 +43,42 @@ export async function getProdTransactionsFromDb(productType: string): Promise<ap
 
 /**
  * Connector for the API and database to create a new product transaction
+ * Throws error on invalid request
+ * Throws eror with message "unhandled error" for unhandled errors
  */
 export async function createProdTransactionsInDb(apiReq: api.ProdTransactionReq): Promise<string> {
-  const productTypeIdResult = await db.getProductId(apiReq.productType);
+  let productTypeId: number = -1;
+  const invalidFields: string[] = [];
+
+  // validate request
```

Comment:
no big deal but maybe it'll be worth while putting this in a function? so then the update function can use it too to validate the fields